packages/fast-components-react-base/src/slider/slider.tsx

```diff
@@ -0,0 +1,1078 @@
+                value={formattedValue}
+                onChange={this.onInputValueChange}
+                disabled={this.props.disabled || null}
+                style={{
+                    display: "none",
+                }}
+            />
+        );
+    }
+
+    /**
+     * This exists only to suppress a build warning
+     */
+    private onInputValueChange = (event: React.ChangeEvent): void => {
+        return null;
+    };
+
+    /**
+     * Handles track clicks
+     */
+    private handleTrackMouseDown = (event: React.MouseEvent): void => {
+        if (
+            this.props.disabled ||
+            event.defaultPrevented ||
+            this.state.isDragging ||
+            this.state.isIncrementing
+        ) {
+            return;
+        }
+        event.preventDefault();
+        this.updateDirection();
+        this.updateSliderDimensions();
+        const pixelCoordinate: number =
+            this.props.orientation === SliderOrientation.vertical
+                ? event.pageY
+                : event.pageX;
+        const newValue: number =
+            (this.props.range.maxValue - this.props.range.minValue) *
+                this.convertPixelToPercent(pixelCoordinate) +
+            this.props.range.minValue;
+
+        switch (this.props.mode) {
+            case SliderMode.singleValue:
+                this.updateValues(newValue, newValue);
+                break;
+
+            case SliderMode.adjustBoth:
+                this.handleMultiThumbTrackClick(newValue);
+                break;
+
+            case SliderMode.adustLowerValue:
+                this.updateValues(newValue, null);
+                break;
+
+            case SliderMode.adustUpperValue:
+                this.updateValues(null, newValue);
+                break;
+        }
+    };
+
+    /**
+     *  Handles track clicks when there are multiple thumbs
+     */
+    private handleMultiThumbTrackClick = (value: number): void => {
+        if (value <= this.state.lowerValue) {
+            this.updateValues(value, null);
+        } else if (value >= this.state.upperValue) {
+            this.updateValues(null, value);
+        } else {
+            // between values move the closest thumb to the click
+            if (value - this.state.lowerValue < this.state.upperValue - value) {
+                this.updateValues(value, null);
+            } else {
+                this.updateValues(null, value);
+            }
+        }
+    };
+
+    /**
+     *  Updates the direction
+     */
+    private updateDirection = (): void => {
+        if (this.rootElement.current === null) {
+            return;
+        }
+
+        const closest: Element = this.rootElement.current.closest(`[dir]`);
+
+        if (closest === null) {
+            this.direction = "ltr";
+            return;
+        }
+
+        const foundDir: string = closest.getAttribute("dir");
+
+        if (foundDir === "ltr" || foundDir === "rtl") {
```

Comment:
there should be an enum you can use in web utilities

Reply:
yep